liquidator/OneInchExchange.js

```diff
@@ -0,0 +1,73 @@
+const Token = artifacts.require("ExpandedERC20");
+
+const oneSplitAbi = require("./abi/OneSplit.json");
+const oneSplitAddress = "0xC586BeF4a0992C495Cf22e1aeEE4E446CECDee0E";
+
+// As defined in 1inch
+const ETH_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
+
+class OneInchExchange {
+  /**
+   * @notice Creates OneInchExchange client
+   * @param {Object} web3 Web3 instance
+   * */
+  constructor({ web3, gasEstimator }) {
```

Comment:
Just curious -- why did you decide to put these in an object rather than a param list? (I personally like this approach, but just wanted to hear the reasoning, since it's somewhat uncommon)

Reply:
I guess you can count this as premature optimization right now, but I'm assuming we might have additional dependencies that we would like to pass into the constructor into the future as some point (for example the logger).

I personally prefer this approach as in the long run, it makes the arguments you supply a lot more readable i.e.

```javascript
const t = new Object(10, 20, 30, false, true)

// vs

const t = new Object({
    delay: 10,
    maxRetries: 20,
    hops: 30,
    logging: false,
    retry: true
})
```

Ultimately this is more art than science, and am happy to change it to a param list if you'd prefer it that way.

Reply:
Nope, I like this way better. +1. I think it makes adding arguments anywhere in the list easier, it means you don't depend on ordering, and it means that you always provide arguments by name.

Was just curious to hear why you prefer it!